src/poetry/factory.py

```diff
@@ -178,23 +180,28 @@ def create_legacy_repository(
         )
 
     @classmethod
-    def create_pyproject_from_package(cls, package: ProjectPackage, path: Path) -> None:
+    def create_pyproject_from_package(
+        cls, package: Package, path: Path | None = None
+    ) -> TOMLDocument:
         import tomlkit
 
-        from poetry.layouts.layout import POETRY_DEFAULT
-
-        pyproject = tomlkit.loads(POETRY_DEFAULT)
-        content = pyproject["tool"]["poetry"]
+        pyproject = tomlkit.document()
+        pyproject["tool"] = tomlkit.table()
+        pyproject["tool"]._is_super_table = True
```

Comment:
```suggestion
        pyproject["tool"] = tomlkit.table(is_super_table=True)
```

Analoguous for other super tables.

Reply:
Won't work because of a bug in the import order. The poetry-core vendored dependency takes precedence and it's an older version at present. Will have to wait until next core version for this change.